src/inference/src/cpp_interfaces/interface/ie_iinfer_request_internal.cpp

```diff
@@ -157,6 +158,123 @@ void IInferRequestInternal::SetBlob(const std::string& name, const Blob::Ptr& us
+        OPENVINO_ASSERT(item->getTensorDesc().getDims()[batch_idx] == 1,
+                        "set_input_tensors/set_tensors. Tensors shall represent one item in a batch, ",
+                        item->getTensorDesc().getDims()[batch_idx],
+                        " provided");
+        return res + item->getTensorDesc().getDims()[batch_idx];
+    });
+    if (param->get_partial_shape().rank().is_static()) {
+        OPENVINO_ASSERT(batch_idx >= 0 && batch_idx < param->get_partial_shape().rank().get_length(),
+                        "set_input_tensors/set_tensors error. Layout ",
+                        param->get_layout().to_string(),
+                        " is incorrect for input '",
+                        name,
+                        "' with shape ",
+                        param->get_partial_shape());
+        auto batch = param->get_partial_shape()[batch_idx];
+
+        OPENVINO_ASSERT(batch.is_dynamic() || batch.get_length() == input_size,
+                        "set_input_tensors/set_tensors error. Input shape ",
+                        param->get_partial_shape(),
+                        "batch ",
+                        batch,
+                        "doesn't match with total blobs count: ",
+                        input_size);
+    }
+    OPENVINO_ASSERT(batch_idx == 0,
+                    "set_input_tensors/set_tensors is not currently supported for batch dimension index ",
+                    batch_idx,
+                    " != 0");
+    // Initial implementation always performs creation of new Blob and 'memcpy'
+    // In future consider checking if blobs point to contiguous range of memory
+    auto& desc = blobs[0]->getTensorDesc();
+    desc.getDims()[batch_idx] = input_size;
+    auto blockingDims = desc.getBlockingDesc().getBlockDims();
+    blockingDims[batch_idx] = input_size;
+    auto blockingDesc = BlockingDesc(blockingDims,
+                                     desc.getBlockingDesc().getOrder(),
+                                     desc.getBlockingDesc().getOffsetPadding(),
+                                     desc.getBlockingDesc().getOffsetPaddingToData(),
+                                     desc.getBlockingDesc().getStrides());
+    auto batched_desc = InferenceEngine::TensorDesc(desc.getPrecision(), desc.getDims(), blockingDesc);
+    auto desc_to_string = [](const TensorDesc& desc) {
+        std::stringstream s;
+        s << "{ " << desc.getLayout() << " " << desc.getPrecision().name();
+        s << "dim=(";
+        for (const auto& d : desc.getDims()) {
+            s << " " << d;
+        }
+        s << " ) }";
+        return s.str();
+    };
+    std::for_each(blobs.begin(), blobs.end(), [&desc, &batch_idx, &desc_to_string](const Blob::Ptr& item) {
+        auto item_desc = item->getTensorDesc();
+        item_desc.getDims()[batch_idx] = desc.getDims()[batch_idx];
```

Comment:
suppose it's not needed. Let's just compare `desc` (blobs[0]'s desc) and current desc